vault/dynamic_system_view.go

```diff
@@ -479,3 +479,12 @@ func (d dynamicSystemView) GenerateIdentityToken(ctx context.Context, req *plugi
 		TTL:   ttl,
 	}, nil
 }
+
+func (d dynamicSystemView) HasLicense() (bool, error) {
+	licenseState, err := d.core.EntGetLicenseState()
```

Comment:
On the one hand, I like this approach because it is simple. It does not require us to update https://github.com/hashicorp/vault-licensing and other places in Vault core's code to add a new ent plugin. 

On the other hand, it sidesteps the current paradigm of explicitly allowlisting each feature/plugin. So I am not sure about this. I am having trouble coming up with any downsides to your current approach other than we don't have fine-grained control. Maybe we should get more feedback from our team?

Reply:
> it sidesteps the current paradigm of explicitly allowlisting each feature/plugin
I agree. Along this line, I wonder, if any point we have different types/tiers of licensing, `HasLicense()` will need to be changed. This makes me think `HasFeature()` of `SystemView` interface, currently left unimplemented, could solve the fine-grained control.